app/src/pointing/input_processor_behaviors.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_input_processor_behaviors
+
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <drivers/input_processor.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/keymap.h>
+#include <zmk/behavior.h>
+#include <zmk/virtual_key_position.h>
+
+struct ip_behaviors_config {
+    uint8_t index;
+    size_t size;
+    uint16_t type;
+
+    const uint16_t *codes;
+    const struct zmk_behavior_binding *bindings;
+};
+
+static int ip_behaviors_handle_event(const struct device *dev, struct input_event *event,
+                                     uint32_t param1, uint32_t param2,
+                                     struct zmk_input_processor_state *state) {
+    const struct ip_behaviors_config *cfg = dev->config;
+
+    if (event->type != cfg->type) {
+        return 0;
+    }
+
+    for (size_t i = 0; i < cfg->size; i++) {
+        if (cfg->codes[i] == event->code) {
+            struct zmk_behavior_binding_event behavior_event = {
+                .position = ZMK_VIRTUAL_KEY_POSITION_BEHAVIOR_INPUT_PROCESSOR(
+                    state->input_device_index, cfg->index),
+                .timestamp = k_uptime_get(),
+#if IS_ENABLED(CONFIG_ZMK_SPLIT)
+                .source = ZMK_POSITION_STATE_CHANGE_SOURCE_LOCAL,
```

Comment:
Is there a way to associate the listener with the pointing device that creates the input event? It could let us assign a locality, at least if we assume one pointing device per listener (though that is an incorrect assumption, I think?)

If that's not possible or something we don't want to tackle yet, we should add a note like this one that it always triggers on central: https://zmk.dev/docs/keymaps/combos#advanced-usage

Reply:
Yeah, I can note it only ever fires on the central. Tracking/identifying will require more work, and I'm not sure it's worth it this first go round.

Reply:
Noted.